client/service/src/task_manager/mod.rs

```diff
@@ -336,9 +336,12 @@ impl TaskManager {
 		}
 	}
 
-	/// Set what the task manager should keep alive.
-	pub(super) fn keep_alive<T: 'static + Send + Sync>(&mut self, to_keep_alive: T) {
-		self.keep_alive = Box::new(to_keep_alive);
+	/// Set what the task manager should keep alive, can be called multiple times.
+	pub fn keep_alive<T: 'static + Send + Sync>(&mut self, to_keep_alive: T) {
+		// allows this fn to safely called multiple times.
+		use std::mem;
+		let old = mem::replace(&mut self.keep_alive, Box::new(()));
+		self.keep_alive = Box::new((to_keep_alive, old));
```

Comment:
If you _absolutely need to_ call this function more than once, then this is fine..

Reply:
It's just to tempting to call it more than once imo. If there is no downside we should do it